src/Core/Grid/Data/Factory/AttachmentGridDataFactoryDecorator.php

```diff
@@ -0,0 +1,160 @@
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
+
+use Doctrine\DBAL\Connection;
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShop\PrestaShop\Core\Util\File\FileSizeConverter;
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+
+/**
+ * Decorates attachment grid data factory
+ */
+final class AttachmentGridDataFactoryDecorator implements GridDataFactoryInterface
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $attachmentDoctrineGridDataFactory;
+
+    /**
+     * @var int
+     */
+    private $employeeIdLang;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var FileSizeConverter
+     */
+    private $fileSizeConverter;
+
+    /**
+     * @param GridDataFactoryInterface $attachmentDoctrineGridDataFactory
+     * @param int $employeeIdLang
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param FileSizeConverter $fileSizeConverter
+     */
+    public function __construct(
+        GridDataFactoryInterface $attachmentDoctrineGridDataFactory,
+        int $employeeIdLang,
+        Connection $connection,
+        string $dbPrefix,
+        FileSizeConverter $fileSizeConverter
+    ) {
+        $this->attachmentDoctrineGridDataFactory = $attachmentDoctrineGridDataFactory;
+        $this->employeeIdLang = $employeeIdLang;
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+        $this->fileSizeConverter = $fileSizeConverter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $attachmentData = $this->attachmentDoctrineGridDataFactory->getData($searchCriteria);
+
+        $attachmentRecords = $this->applyModifications($attachmentData->getRecords());
+
+        return new GridData(
+            $attachmentRecords,
+            $attachmentData->getRecordsTotal(),
+            $attachmentData->getQuery()
+        );
+    }
+
+    /**
+     * @param RecordCollectionInterface $attachments
+     *
+     * @return RecordCollection
+     */
+    private function applyModifications(RecordCollectionInterface $attachments): RecordCollection
+    {
+        $modifiedAttachments = [];
+
+        foreach ($attachments as $attachment) {
+            if ((int) $attachment['products'] > 0) {
```

Comment:
Count missing?
Are we returning products, or productIds?

Reply:
(Because of `products_names` below

Reply:
products in this case is integer value of products count returned from query builder and not the result of product_names query :)